### PR TITLE
GraphQL Auth Endpoints

### PR DESCRIPTION
**Pull request by thetre97**
Have added in a graphQL server for each auth endpoint, similar to the seemingly abandoned #18 .

### CODE REVIEW

**Comment by thetre97** [comment]
A graphql endpoint could be useful in a coupleof cases, for example I frequently use https://github.com/nuxt-community/apollo-module, which has a helper for adding a  token (and refreshing cache etc), so I can just fetch the token using apollo, and store the token - rather than having to install axios, nuxt-auth etc.

**Comment by elitan** [comment]
Very nice @thetre97 . But I would like to learn more about the use case.

I do not have much experience with Vue. What I did with React was to build a [nhost-js-sdk](https://github.com/nhost/nhost-js-sdk) do handle the interaction with hasura-backend-plus.

Would a js-sdk such as nhost-js-sdk work for you too in Vue?

**Comment by thetre97** [comment]
[https://tr.cloudmagic.com/h/v6/emailtag/tag/2.0/1566491090/0a337f7a278d0a08a46115af74d0fa27/35/d1aaeab55557e98a92b1cd798143c9b5/d56938a26abbc09d9f082585346092bb/783753f3d9f7c59636403c4719987da0/newton.gif]

No problem - I'll try and explain it a bit more.


As I mentioned, I use nuxt quite a bit, and the current nhost js SDK wouldn't work very well with it, as (at least, the last time I looked) it stores the token in localhost -  So, when I am making Apollo requests server side (prefetching), it obviously doesn't have access to the browser storage. This could be fairly easily solved by using 'universal-cookies' or something similar. I'd be happy to help create a Vue/nuxt 'version' if it would be easier...


My reason for adding a graphql endpoint: I use nuxt-apollo, which integrates vue-apollo into nuxt - This package provides a helper `this.$apolloHelpers.onLogin([accessToken])` which you can use to store the jwt token in a cookie, so when prefetching Apollo requests the nuxt server can read the cookie, and get the jwt token.

A graphql endpoint would be useful for me, because I could just run an Apollo mutation on login, then store that token (or the refetch token) with that helper.


Admittedly, it's not essential, as I could just use axios to post data, then use the helper again to store the token. That just means adding an extra library which I only use once in my app...


Hope that makes sense?

On Thu, Aug 22, 2019 at 4:55pm, Johan Eliasson <notifications@github.com<mailto:notifications@github.com>> wrote:


Very nice @thetre97<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fthetre97&data=02%7C01%7C%7C624764e27e394f83449908d727191e91%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637020861145740769&sdata=RTNr8m4Va7wZjzonrC9n7S0zJYwt5MDXin9njVFdkAo%3D&reserved=0> . But I would like to learn more about the use case.

I do not have much experience with Vue. What I did with React was to build a nhost-js-sdk<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnhost%2Fnhost-js-sdk&data=02%7C01%7C%7C624764e27e394f83449908d727191e91%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637020861145740769&sdata=nIOp8B03f%2BHx3hmAbWbUNtdQuzXUwdzZh49WXbg2I70%3D&reserved=0> do handle the interaction with hasura-backend-plus.

Would a js-sdk such as nhost-js-sdk work for you too in Vue?

File: src/graphql/schema/resolvers.js [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+  USER_MANAGEMENT_DATABASE_SCHEMA_NAME,
+  REFETCH_TOKEN_EXPIRES,
+  HASURA_GRAPHQL_JWT_SECRET,
+} = require('../../config');
+
+const schema_name = USER_MANAGEMENT_DATABASE_SCHEMA_NAME === 'public' ? '' : USER_MANAGEMENT_DATABASE_SCHEMA_NAME.toString().toLowerCase() + '_';
+
+const resolvers = {
+  Query: {
+    hello: () => 'Hello world!',
```
Comment (elitan) [inline src/graphql/schema/resolvers.js]: not needed?

File: src/graphql/schema/typedefs.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+const { gql } = require('apollo-server-express');
+
+const typeDefs = gql`
+  scalar Json
+
+  type Query {
+    hello: String
```
Comment (MarcelloTheArcane) [inline src/graphql/schema/typedefs.js]: What looks like another 'hello world' here too

File: src/graphql/schema/resolvers.js (hunk 1 above)
Comment (thetre97) [reply src/graphql/schema/resolvers.js]: Fixed

**Pull request closed without merging**